src/app/trainee/state/revalidation-notes.state.ts

```diff
@@ -0,0 +1,56 @@
+import { State, Action, StateContext, Selector } from "@ngxs/store";
+import { GetRevalidationNotes } from "./revalidation-notes.actions";
+import { HttpParams } from "@angular/common/http";
+import { notesResponse1 } from "../mock-data/trainee-spec-data";
+import { Injectable } from "@angular/core";
+import { tap, catchError } from "rxjs/operators";
+import { of } from "rxjs";
+import { RevalidationNotesService } from "../services/revalidation-notes.service";
+import { INote } from "../revalidation-history.interface";
+
+export class RevalidationNotesStateModel {
+  public items: INote[];
+}
+
+@State<RevalidationNotesStateModel>({
+  name: "revalidationNotes",
+  defaults: {
+    items: []
+  }
+})
+@Injectable()
+export class RevalidationNotesState {
+  constructor(private service: RevalidationNotesService) {}
+
+  @Selector()
+  public static revalidationNotes(state: RevalidationNotesStateModel) {
+    return state.items;
+  }
+
+  @Action(GetRevalidationNotes, { cancelUncompleted: true })

```

Comment:
for my own understanding can i ask what is `{ cancelUncompleted: true }` and what it does?

Reply:
From my understanding this was a feature asked for in gitHub by devs. This doesn't allow multiple calls to this action i.e. last call is executed and previous calls terminated. 
`https://www.ngxs.io/advanced/cancellation` for notes I will be dispatching updates as users type and patching the state so they don't lose data (only if they refresh the screen they lose) and until they hit save, send to updated value to backend.

Reply:
oh i see, thats really cool. thanks for explaining